WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt

```diff
@@ -731,18 +735,13 @@ class ProductDetailViewModel @Inject constructor(
         else string.product_detail_save_product_success
 
     private fun pickAddProductRequestSnackbarText(productWasAdded: Boolean, requestedProductStatus: ProductStatus) =
-        if (productWasAdded) {
-            if (requestedProductStatus == DRAFT) {
-                string.product_detail_publish_product_draft_success
-            } else {
-                string.product_detail_publish_product_success
-            }
-        } else {
-            if (requestedProductStatus == DRAFT) {
-                string.product_detail_publish_product_draft_error
-            } else {
-                string.product_detail_publish_product_error
-            }
+        when {
+            productWasAdded && requestedProductStatus == DRAFT -> string.product_detail_publish_product_draft_success
+            productWasAdded && requestedProductStatus == PUBLISH -> string.product_detail_publish_product_success
+            productWasAdded -> string.product_detail_save_product_success
+            !productWasAdded && requestedProductStatus == DRAFT -> string.product_detail_publish_product_draft_error
+            !productWasAdded && requestedProductStatus == PUBLISH -> string.product_detail_publish_product_error
+            else -> string.product_detail_save_product_error
```

Comment:
np: I have a feeling that this is somewhat harder to read than before, the subtle `!` negation combined with the predicate composition with repetitions of the same variables makes all scenarios compressed with each other, the previous way looked to be easier to understand how to handle every scenario. Maybe you could do an `if / else` for the `productWasAdded` verification and inner when for the ProductStatus? WDYT?

```suggestion
        if (productWasAdded) {
            when(requestedProductStatus) {
                DRAFT -> string.product_detail_publish_product_draft_success
                PUBLISH -> string.product_detail_publish_product_success
                else -> string.product_detail_save_product_success
            }
        } else {
            when(requestedProductStatus) {
                DRAFT -> string.product_detail_publish_product_draft_error
                PUBLISH -> string.product_detail_publish_product_error
                else -> string.product_detail_save_product_error
            }
        }
```

Reply:
Great work @ThomazFB. It is always interesting to have this kind of feedback to see how others approach the same problem, for me is always a learning experience. Thanks! 🙏🏻

>I have a feeling that this is somewhat harder to read than before, the subtle ! negation combined with the predicate composition with repetitions of the same variables makes all scenarios compressed with each other, the previous way looked to be easier to understand how to handle every scenario.

I totally agree with you here. The `!` is subtle, and the code could be hard to read combined with the predicate composition. 

However, I'm not 100% convinced about having an `if` and a nested `when`. This is a personal preference and a very subjective opinion.
That's the reason why I came up with this 3rd variation, and I would love to hear if you feel that this one somehow is easy to read. WDYT?
```
    private fun pickAddProductRequestSnackbarText(productWasAdded: Boolean, requestedProductStatus: ProductStatus) {
        val isDraftStatus = requestedProductStatus == DRAFT
        val isPublishStatus = requestedProductStatus == PUBLISH
        val failedAddingProduct = !productWasAdded
        return when {
            productWasAdded && isDraftStatus -> string.product_detail_publish_product_draft_success
            productWasAdded && isPublishStatus -> string.product_detail_publish_product_success
            failedAddingProduct && isDraftStatus -> string.product_detail_publish_product_draft_error
            failedAddingProduct && isPublishStatus -> string.product_detail_publish_product_error
            productWasAdded -> string.product_detail_save_product_success
            else -> string.product_detail_save_product_error
        }

    }

```

Reply:
Sounds great this way, @atorresveiga! Thanks for taking some time to rethink the readability of this method! I agree with your opinion on this one, go ahead with your suggestion!

Reply:
Great @ThomazFB! 🙌 
Fixed here 78ade6c